### PR TITLE
Prover documentation

### PR DESCRIPTION
**Pull request by irakliyk**
This PR will add documentation to `prover` crate in preparation for publishing to crates.io.

### CODE REVIEW

File: prover/Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,7 @@ fri = { path = '../fri', package = "winter-fri" }
 air = { path = "../air", package = "winter-air" }
 rayon = { version = "1.5", optional = true }
 log = "0.4"
+
+# Allow math in docs
+[package.metadata.docs.rs]
+rustdoc-args = ["--html-in-header", ".cargo/katex-header.html"]
```
Comment (kchalkias) [inline prover/Cargo.toml:30]: newline

**Review by kchalkias — APPROVED** [review]
Approved, I would prefer an extra pair of eyes in prover's `lib.rs`

File: prover/src/lib.rs [hunk 2]
```diff
@@ -3,23 +3,359 @@
...
+    // the verifier. the trace polynomials are actually evaluated over two points: z and z * g,
+    // where g is the generator of the trace domain.
+    let ood_frame = trace_polys.get_ood_frame(z);
+    channel.send_ood_evaluation_frame(&ood_frame);
+
+    let ood_evaluations = composition_poly.evaluate_at(z);
+    channel.send_ood_constraint_evaluations(&ood_evaluations);
+
+    // draw random coefficients to use during DEEP polynomial composition, and use them to
+    // initialize the DEPP composition polynomial
```
Comment (Jasleen1) [inline prover/src/lib.rs:279]: nit: typo in DEEP

File: prover/src/lib.rs [hunk 3]
```diff
@@ -3,23 +3,359 @@
...
+// ================================================================================================
+/// Returns a STARK proof attesting to a correct execution of a computation.
+///
+/// Function parameters have the following meanings:
+/// * `AIR` is a type implementing [Air] trait for the computation. Among other things, it defines
+///    algebraic constraints which define the computation.
+/// * `trace` is an execution trace of the computation executed against some set of inputs. These
+///   inputs may include both public and private inputs.
+/// * `pub_inputs` is the set of public inputs against which the computation was executed. These
+///   these inputs will beed to be shared with the verifier in order for them to verify the proof.
```
Comment (Jasleen1) [inline prover/src/lib.rs:100]: nit: typo `beed` instead of `need`

**Review by Jasleen1 — APPROVED** [review]

File: prover/src/lib.rs (hunk 3 above)
Comment (irakliyk) [reply prover/src/lib.rs]: Thank you! fixed!

File: prover/src/lib.rs (hunk 2 above)
Comment (irakliyk) [reply prover/src/lib.rs]: Thank you! fixed!

File: prover/Cargo.toml (hunk 1 above)
Comment (irakliyk) [reply prover/Cargo.toml]: Thank you! fixed!

**Pull request merged**